### PR TITLE
feat(fgrid):  Add alternative to dgrid

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: demos/gallery/.vscode/launch.json [hunk 1]
```diff
@@ -12,11 +12,22 @@
...
+                "--disable-web-security"
+            ],
+            "webRoot": "${workspaceRoot}"
+        },
         {
             "name": "galleryItem",
             "type": "chrome",
             "request": "launch",
-            "url": "file:///${workspaceRoot}/galleryItem.html?./samples/geospatial/Leaflet/Countries.js",
+            "url": "file:///${workspaceRoot}/galleryItem.html?./samples/layout/Chart Panel.js",
```
Comment (jeclrsg) [inline demos/gallery/.vscode/launch.json:30]: Extra space in filename?

Comment (GordonSmith) [reply demos/gallery/.vscode/launch.json:30]: Its correct - there is a space in the filename

File: packages/fgrid/src/DBStore.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                ...formattedRow,
+                __hpcc_id: opts.start + idx,
+                __origRow: row
+            };
+        });
+    }
+
+    fetchRange(opts: { start: number, end: number }): Promise<object[]> {
+        const data = this._fetchRange(opts);
+        const retVal: any = new Deferred();
```
Comment (jeclrsg) [inline packages/fgrid/src/DBStore.ts:72]: Could this be Promise or is it for a backwards compatibility with dgrid or something?

File: packages/fgrid/src/table.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+import { useCallback, useEffect, useMemo, useState } from "preact/hooks";
+import { unmountComponentAtNode } from "preact/compat";
+import { DetailsList, DetailsListLayoutMode, initializeIcons, IColumn, IDetailsHeaderProps, IObjectWithKey, Selection, SelectionMode, TooltipHost, IDetailsListStyles, IRenderFunction, IDetailsColumnRenderTooltipProps, ConstrainMode } from "@fluentui/react";
+
+initializeIcons();
+
+export type QuerySortItem = { attribute: string, descending: boolean };
+function copyAndSort<T>(items: T[], attribute: string, descending?: boolean): T[] {
+    const key = attribute as keyof T;
+    return [...items].sort((a: T, b: T) => ((descending ? a[key] < b[key] : a[key] > b[key]) ? 1 : -1));
```
Comment (jeclrsg) [inline packages/fgrid/src/table.tsx:12]: Should there be a case for a == b?

File: packages/fgrid/src/table.tsx [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+    protected _div;
+
+    constructor() {
+        super();
+    }
+
+    _prevRow;
+    renderTable() {
+        // return <h1>Hello</h1>;
```
Comment (jeclrsg) [inline packages/fgrid/src/table.tsx:190]: commented out test code

File: packages/fgrid/src/table.tsx [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+            this._prevRow = row;
+        }} />;
+    }
+
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        this._div = element
+            .append("div")
+            .style("margin-right", "18px")
+            // .style("display", "inline-block")
```
Comment (jeclrsg) [inline packages/fgrid/src/table.tsx:207]: and these two?

File: packages/fgrid/src/DBStore.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/fgrid/src/DBStore.ts:72]: Actually I should delete this file until we actually need it.

**Review by jbrundage — APPROVED** [review]

**Pull request merged**